File: src/217_contains_duplicate/contains_duplicate.py
class Solution:
    def containsDuplicate(self, nums):
        seen = set()

        for num in nums:
            if num in seen:
                return True

            seen.add(num)

        return False


s = Solution()
nums = [1, 2, 3, 1]
print(s.containsDuplicate(nums), True)
nums = [1, 2, 3, 4]
print(s.containsDuplicate(nums), False)
nums = [1, 1, 1, 3, 3, 4, 3, 2, 4, 2]
print(s.containsDuplicate(nums), True)

File: src/217_contains_duplicate/ContainsDuplicate.java
import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
    public boolean containsDuplicate(int[] nums) {
        Set<Integer> seen = new HashSet<>();

        for (int num : nums) {
            if (seen.contains(num))
                return true;

            seen.add(num);
        }
        return false;
    }

    public static void main(String[] args) {
        ContainsDuplicate s = new ContainsDuplicate();
        int[] nums1 = { 1, 2, 3, 1 };
        System.out.println(s.containsDuplicate(nums1) + " true");
        int[] nums2 = { 1, 2, 3, 4 };
        System.out.println(s.containsDuplicate(nums2) + " false");
        int[] nums3 = { 1, 1, 1, 3, 3, 4, 3, 2, 4, 2 };
        System.out.println(s.containsDuplicate(nums3) + " true");
    }
}
